content/documentation/staging/tutorial/04-observe/01-enable-sidecars-for-all-namespaces.adoc

```diff
@@ -0,0 +1,59 @@
+---
+title: Enable Sidecars in all workloads
+weight: 40
+---
+:icons: font
+:sectlinks:
+
+Istio sidecar proxy allows to add a workload into the mesh.
+
+Proxies connect with the control plane and provide link:https://istio.io/latest/docs/concepts/what-is-istio/#why-use-istio[Service Mesh functionality].
+
+One of the main features is the automatic metrics, logs and traces provide automatically by the sidecar.
+
+In the previous steps we have added a sidecar only in the *control* workload from *travel-control* namespace.
+
+We have added new powerful features but application are still missing visibility from other workloads.
+
+NOTE: *[1]* Select multiple namespaces to identify missing sidecars in the Travel Demo application
+++++
+<a class="image-popup-fit-height" href="/images/tutorial/04-01-missing-sidecars.png" title="Missing Sidecars">
+    <img src="/images/tutorial/04-01-missing-sidecars.png" style="display:block;margin: 0 auto;" />
+</a>
+++++
+
+{nbsp} +
+That *control* workload will give a good visibility of its traffic but telemetry is partially enabled as *travel-portal* and *travel-agency* workloads don't have sidecar proxies.
+
+NOTE: *[2]* Enable proxy injection in *travel-portal* and *travel-agency* namespaces
+
+In the First Steps of this tutorial we didn't inject the sidecar proxies on purpose to show an scenario where only some workloads may have sidecars.
+
+Typically, Istio users annotate namespace before the deployment to allow Istio to automatically add the sidecar when the application is rolled out into the cluster.
+
+[source,bash]
+----
+kubectl label namespace travel-agency istio-injection=enabled
+kubectl label namespace travel-portal istio-injection=enabled
+
+kubectl delete -f <(curl -L https://raw.githubusercontent.com/kiali/demos/master/travels/travel_agency.yaml) -n travel-agency
```

Comment:
instead of removing / resending everything we could also just delete the pods, this is perhaps more straightforward:

```
kubectl delete pods --all -n travel-agency
kubectl delete pods --all -n travel-portal
```

Reply:
I'm going to test, but not sure if if works if the controller is not update the Istio webhook chain may not known that it needs to inject a new sidecar.

But I guess if you update the controller, then the webhook checks if namespace label is present.

Reply:
Yes, I'm afraid this mechanism works as I described.

```
1.- Anotate the namespace with the
2.- Create the controller
3.- The controller generates the pod with the sidecar
```

Existing controllers needs to be removed (or updated) but as annotation lives at namespace level, it doesn't help to just annotate the namespace + delete the pods, then the controller is not taking the changes from the Webhook chain.

Reply:
hmm i've always just deleted pods, and it's always worked for me (not just for this demo, but in general)